content/lessons/hooks/intro-to-react-hooks.md

```diff
@@ -93,22 +94,24 @@ preReadQuiz:
       explanation: |
         Snippet 1 will result in a JavaScript syntax error. Snippet 2, while not incorrect, does not provide an initial value to 'useState', which will produce an initial value of 'count' as 'undefined'. When trying to increment the value in 'handleButtonClick', incrementing 'count' by one will return 'NaN'. Snippet 3 is missing an array destructuring statement, and Snippet 5 attempts to import the functionality of useState as if it could be imported, when instead it must be array-destructured. 
       type: radio
-    - description: True or False - React Hooks give you permission to refactor all of your Class Components to be Hooked Components.
+    - description: True or False - You can update state (with a function returned from useState) within a useEffect hook.
       choices:
       - value: 'True'
       - value: 'False'
       correctChoices: 
       - 1
-      explanation: React Hooks are best implemented in new code, or to move away from unsafe/unstable patterns. Class Components are still 100% valid, and the time invested in refactoring old code is better spent in writing new code with Hooks.
+      explanation: You can do this, but remember to specify values in the second argument to useEffect. The hook will then only run when any value in the second argument changes. If you don't, you will enter into an infinite loop (as changing the state triggers a re-render, which calls the useEffect hook.)
       type: radio
     - description: When you return a function from within a useEffect call, when does this function get executed?
       choices:
       - value: When the component mounts
-      - value: When the component unmounts
       - value: When the component's children are mounted
+      - value: When the component unmounts
       - value: When the component throws an error
+      - value: When the component re-renders
```

Comment:
I think this option should be "When a value in useEffect's dependency array changes," since it won't be executed when the component re-renders if none of those values change.

Reply:
Great callout, will change!